cypress/integration/pages/frontPage.js

```diff
@@ -0,0 +1,167 @@
+import iterator from '../../support/iterator';
+import envConfig from '../../support/config/envs';
+import config from '../../support/config/services';
+import appConfig from '../../../src/app/lib/config/services';
+
+const tests = ({ service }) =>
+  describe(`Tests`, () => {
+    describe('Frontpage body', () => {
+      before(() => {
+        cy.viewport(1008, 768);
+      });
+
+      describe('Header', () => {
+        if (appConfig[service].navigation) {
+          it('should have one visible navigation', () => {
+            cy.get('nav')
+              .should('have.lengthOf', 1)
+              .should('be.visible');
+          });
+        }
+
+        it('should have a visually hidden top-level header', () => {
+          cy.get('h1').should('have.length', 1);
+        });
+      });
+
+      describe('Section', () => {
+        it('should be labelled by a visible section label', () => {
+          cy.get('section')
+            .should('have.length.of.at.least', 1)
+            .should('be.visible')
+            .each($section => {
+              cy.wrap($section).within(() => {
+                cy.get('h2').should('have.lengthOf', 1);
+              });
+            });
+        });
+
+        it('should contain at least one story promo', () => {
+          cy.get('section').within(() => {
+            cy.get('img')
+              .should('have.length.of.at.least', 1)
+              .should('be.visible');
+            cy.get('h3')
+              .should('have.length.of.at.least', 1)
+              .should('be.visible')
+              .find('a')
+              .should('have.attr', 'href');
+            cy.get('p')
+              .should('have.length.of.at.least', 1)
+              .should('be.visible');
+            cy.get('time')
+              .should('have.length.of.at.least', 1)
+              .should('be.visible');
+          });
+        });
+      });
+    });
+  });
+
+// -------------------------------------------
+
+const canonicalOnlyTests = ({ service }) =>
+  describe(`Canonical Tests`, () => {
+    describe('ATI', () => {
+      it('should have a noscript tag with an 1px image with the ati url', () => {
+        cy.hasNoscriptImgAtiUrl(
+          envConfig.atiUrl,
+          config[service].isWorldService ? envConfig.atiAnalyticsWSBucket : '',
+        );
+      });
+
+      it('should not have an AMP attribute', () => {
+        cy.get('html').should('not.have.attr', 'amp');
+      });
+    });
+
+    describe('Scripts', () => {
+      it('should only have expected bundle script tags', () => {
+        cy.hasExpectedJsBundles(envConfig.assetOrigin, service);
+      });
+
+      it('should have 1 bundle for its service', () => {
+        cy.hasOneServiceBundle(service);
+      });
+    });
+  });
+
+// -------------------------------------------
+
+const ampOnlyTests = ({ service }) =>
```

Comment:
since this function actually runs the tests when called what about naming it `runAmpTests` and same for canonical e.g. `runCanonicalTests`